boa3/analyser/typeanalyser.py

```diff
@@ -274,6 +274,70 @@ def get_un_op(self, operator: Operator, operand: Any) -> UnaryOperation:
             expected_type: str = expected_op.operand_type.identifier
             raise CompilerError.MismatchedTypes(0, 0, expected_type, actual_type)
 
+    def visit_Compare(self, compare: ast.Compare) -> Optional[IType]:
+        """
+        Verifies if the types of the operands are valid to the compare operations
+
+        If the operations are valid, changes de Python operator by the Boa operator in the syntax tree
+
+        :param compare: the python ast compare operation node
+        :return: the type of the result of the operation if the operation is valid. Otherwise, returns None
+        :rtype: IType or None
+        """
+        if len(compare.ops) > 1:
+            # TODO: implement when logical operations are implemented
+            raise NotImplementedError
+
+        if len(compare.comparators) != len(compare.ops):
+            self._log_error(
+                CompilerError.IncorrectNumberOfOperands(
+                    compare.lineno,
+                    compare.col_offset,
+                    len(compare.ops),
+                    len(compare.comparators) + 1    # num comparators + compare.left
+                )
+            )
+
+        lineno = compare.lineno
+        col_offset = compare.col_offset
+        try:
+            return_type = None
+            l_operand = self.visit(compare.left)
+            for index, op in enumerate(compare.ops):
+                operator: Operator = self.visit(op)
+                r_operand = self.visit(compare.comparators[index])
+
+                if not isinstance(operator, Operator):
+                    # the operator is invalid or it was not implemented yet
+                    self._log_error(
+                        CompilerError.UnresolvedReference(lineno, col_offset, type(operator).__name__)
+                    )
+
+                operation: BinaryOperation = self.get_bin_op(operator, r_operand, l_operand)
+                if operation is None:
+                    self._log_error(
+                        CompilerError.NotSupportedOperation(lineno, col_offset, operator)
+                    )
+                elif BinaryOp.get_operation(operation) in [BinaryOp.Is, BinaryOp.IsNot, BinaryOp.Eq]:
```

Comment:
Couldn't we add a property on the operation to mark it as "not comparable", I don't know...